SimMuon/MCTruth/src/MuonAssociatorByHitsHelper.cc

```diff
@@ -1197,18 +1198,14 @@ void MuonAssociatorByHitsHelper::getMatchedIds
 	      } else if (printRtS) edm::LogWarning("MuonAssociatorByHitsHelper")
 		<<"*** WARNING in MuonAssociatorByHitsHelper::getMatchedIds, null dynamic_cast of a GEM TrackingRecHit !";
 	      
-	      unsigned int i_detid = (*ithit)->geographicalId().rawId();
-	      GEMDetId i_gemdetid = GEMDetId(i_detid);
-
-	      stringstream i_gem_detector_id;
-	      i_gem_detector_id << i_gemdetid;
+	      if (printRtS){
+		unsigned int i_detid = ithit->geographicalId().rawId();
+		GEMDetId i_gemdetid = GEMDetId(i_detid);
 
-	      stringstream i_ss;
-	      i_ss<<"\t\t hit "<<i_compHit<<" -Muon GEM- detID = "<<i_gem_detector_id.str();
-
-	      string i_hitlog = i_ss.str();
-	      i_hitlog = i_hitlog + write_matched_simtracks(i_SimTrackIds);
-	      if (printRtS) edm::LogVerbatim("MuonAssociatorByHitsHelper") << i_hitlog;
+		string i_hitlog = std::to_string(i_gemdetid);
+		i_hitlog = i_hitlog + write_matched_simtracks(i_SimTrackIds);
+		edm::LogVerbatim("MuonAssociatorByHitsHelper") << i_hitlog;
```

Comment:
I removed i_compHit because it wasn't really needed.
Should i remove all the i_compHit in the gem loop too?